### PR TITLE
add regressions to changelog (NEWS.md)

### PR DESCRIPTION
**Pull request by aplavin**
Following recent slack discussion, https://julialang.slack.com/archives/C67910KEH/p1683570942857959.

Brief summary of that:

> Would it make sense to add known regressions to release notes (NEWS.md)? For 1.9 vs 1.8, I've experienced two performance regressions myself, and seen another one discussed:
> <...>
> Surely there are more, but how can users become aware of them?

> Seems reasonable to me; I would want to know about known regressions in a release if I was considering whether to update or not

References issues: JuliaLang/julia#48612 (mine) JuliaLang/LinearAlgebra.jl#998 (@pablosanjose) JuliaLang/julia#48217 (@Moelf). Feel free to append more, or give more details!

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -200,6 +200,14 @@ Tooling Improvements
...
 * Printing of `MethodError` and methods (such as from `methods(my_func)`) is now prettified and colored consistently
   with printing of methods in stacktraces ([#45069]).
+  
+Regressions
+-----------
+
+* Precompilation takes significantly longer, 2x-3x compilations slowdowns are common
+* Some type-unstable code runs orders of magnitude slower ([#48612])
+* Inplace operations on high-dimensional array views may allocate ([#49332])
+* Compile cache gets excessively invalidated on clusters ([#48217])
```
Comment (IanButterworth) [inline NEWS.md:210]: This is a bug that is being diagnosed currently. I don't think it's right to highlight this here (I can't speak to the other issues above)

File: NEWS.md [hunk 2]
```diff
@@ -200,6 +200,13 @@ Tooling Improvements
 
 * Printing of `MethodError` and methods (such as from `methods(my_func)`) is now prettified and colored consistently
   with printing of methods in stacktraces ([#45069]).
+  
+Regressions
+-----------
+
+* Precompilation takes significantly longer, 2x-3x compilations slowdowns are common
+* Some type-unstable code runs orders of magnitude slower ([#48612])
```
Comment (giordano) [inline NEWS.md:208]: Quoting myself from slack:

> Remember to add the improvement back in the release notes of 1.10 too

**Comment by KristofferC** [comment]
Concretely: https://github.com/JuliaLang/julia/issues?q=+label%3A%22regression+1.9%22+

File: NEWS.md [hunk 3]
```diff
@@ -204,9 +204,7 @@ Tooling Improvements
 Regressions
 -----------
 
-* Precompilation takes significantly longer, 2x-3x compilations slowdowns are common
-* Some type-unstable code runs orders of magnitude slower ([#48612])
-* Inplace operations on high-dimensional array views may allocate and become slower ([#49332])
+[The list](https://github.com/JuliaLang/julia/issues?q=+label%3A%22regression+1.9%22+) of known regressions in the v1.9 release.
```
Comment (KristofferC) [inline NEWS.md:207]:
```suggestion
A list of known regressions in the v1.9 release can be found [here](https://github.com/JuliaLang/julia/issues?q=+label%3A%22regression+1.9%22+)
```

maybe?

File: NEWS.md [hunk 4]
```diff
@@ -200,6 +200,11 @@ Tooling Improvements
 
 * Printing of `MethodError` and methods (such as from `methods(my_func)`) is now prettified and colored consistently
   with printing of methods in stacktraces ([#45069]).
+
```
Comment (IanButterworth) [inline NEWS.md:203]:
```suggestion

```

File: NEWS.md [hunk 5]
```diff
@@ -200,6 +200,10 @@ Tooling Improvements
 
 * Printing of `MethodError` and methods (such as from `methods(my_func)`) is now prettified and colored consistently
   with printing of methods in stacktraces ([#45069]).
+Regressions
```
Comment (IanButterworth) [inline NEWS.md:203]:
```suggestion

Regressions
```

**Pull request merged**